core/commands/storage.go

```diff
@@ -628,20 +632,41 @@ func payFullToEscrow(response *escrowPb.SignedSubmitContractResult, configuratio
 		log.Error(err)
 		return
 	}
+	//TODO talk with Jin for doing signature for every contract
 	// get escrow sig, add them to guard
 	sig := payinRes.EscrowSignature
-	for _, guardContract := range guardContracts {
+	for _, guardContract := range ss.GetGuardContracts() {
 		guardContract.EscrowSignature = sig
 		guardContract.EscrowSignedTime = payinRes.Result.EscrowSignedTime
 		guardContract.LastModifyTime = time.Now()
 	}
 	// TODO: upload filemeta to guard
-	go UploadFileMetaToGuard()
+	go UploadFileMetaToGuard(ss, ss.GetGuardContracts(), response, payerPrivKey, configuration)
 }
 
 // TODO: after upload persist data in leveldb
-func UploadFileMetaToGuard() {
-
+func UploadFileMetaToGuard(ss *storage.FileContracts, contracts []*guardPb.Contract,
+	escrowResults *escrowPb.SignedSubmitContractResult, payerPriKey ic.PrivKey, configuration *config.Config) {
+	fileStatus, err := guard.NewFileStatus(ss, contracts, configuration)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	//TODO below code supposedly will copy the escrow signature from every contract to the guard contract
```

Comment:
check on latest master code, this part has been done before calling uploadFileMeta function, you can take it as input

Reply:
I saw that code, and actually use that code now. And Jin and me agreed to discuss for update this part after demo so that every contract has its designated signature from escrow. So reserve this TODO here.